### PR TITLE
Add support to handle gzipped requests with tests

### PR DESCRIPTION
**Pull request by strefethen**
Added support for gzip to HttpClient.ts which addresses this PR:

https://github.com/microsoft/typed-rest-client/issues/148

I found similar code on SO here and adapted it for this package.

https://stackoverflow.com/questions/12148948/how-do-i-ungzip-decompress-a-nodejs-requests-module-gzip-response-body

### CODE REVIEW

**Comment by strefethen** [comment]
I totally forgot, I need to get this repo reviewed by my employer prior to completing the PR. I'll check back once it's been approved. Sigh.

File: lib/HttpClient.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import https = require("https");
 import tunnel = require("tunnel");
 import ifm = require('./Interfaces');
 import fs = require('fs');
+import zlib = require('zlib');
```
Comment (stephenmichaelf) [inline lib/HttpClient.ts]: The [GitHub Page](https://github.com/kkaefer/DEPRECATED-node-zlib) says this lib is deprecated. We should find another option :)

**Comment by stephenmichaelf** [comment]
Thanks for taking the time to create a PR! I am going to review it then we can discuss again once you have sign off. Appreciate the contribution.

File: lib/HttpClient.ts [hunk 2]
```diff
@@ -51,15 +52,28 @@ export class HttpClientResponse implements ifm.IHttpClientResponse {
...
-            this.message.on('data', (chunk: string) => {
-                output += chunk;
-            });
-
-            this.message.on('end', () => {
-                resolve(output);
-            });
+            if (this.message.headers["content-encoding"] === "gzip") {
+                var buffer = [];
+                let unzipStream = zlib.createGunzip();
```
Comment (stephenmichaelf) [inline lib/HttpClient.ts]: const, add type

**Comment by bryanmacfarlane** [comment]
@strefethen - let me know if you're still interested in pushing this forward with the support that was folded into node.  If not, I can pick this up.

**Comment by strefethen** [comment]
> @strefethen - let me know if you're still interested in pushing this forward with the support that was folded into node. If not, I can pick this up.

@bryanmacfarlane sorry for the delay I'm just circling back to this. It took a long time to get employer approval and I rather forgot about this. I also just noticed the comment above about the deprecation of zlib which I haven't resolved and I'm not sure I'm going to get the time to research other options at this point.

**Comment by strefethen** [comment]
I took another quick look at this and the build above is failing after installing node 4.8.6?? I'm not going to spend any further time debugging this for such an old version of node.

**Comment by stephenmichaelf** [comment]
Closing due to being incomplete.

**Pull request closed without merging**